src/map/Map.java

```diff
@@ -0,0 +1,65 @@
+package map;
+
+import helpers.DataConversionHelper;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Map {
+    private List<Node> nodes;
+
+    public Map() {
+        nodes = new ArrayList<>();
+    }
+
+    public Map(List<Node> nodes) {
+        this.nodes = nodes;
+    }
+
+    public void addNode(Node node) {
+        nodes.add(node);
+    }
+
+    public List<Node> getNodes() {
+        return Collections.unmodifiableList(nodes);
+    }
+
+    public int getIndex(Node n) {
+        int index = nodes.indexOf(n);
+        if (index == -1)
+            throw new IllegalArgumentException("Given Node not found in map.");
+        return index;
+    }
+
+    public int byteSize() {
+        int sum = 2;
+        for (Node n : nodes) {
+            sum += n.byteSize();
+        }
+        return sum;
+    }
+
+    public byte[] toBytes() {

```

Comment:
The methods byteSize() and toBytes() are shared in many classes and seem to fit being from an Interface, not necessary though.

Reply:
yeeeehh no (but yeah)